HeaderLayout.h

```diff
@@ -0,0 +1,52 @@
+#ifndef HEADER_HeaderLayout
+#define HEADER_HeaderLayout
+/*
+htop - HeaderLayout.h
+(C) 2021 htop dev team
+Released under the GNU GPLv2, see the COPYING file
+in the source distribution for its full text.
+*/
+
+#include <assert.h>
+
+#include "Macros.h"
+
+
+typedef enum HeaderLayout_ {
+   HF_TWO_50_50,
+   HF_TWO_33_67,
+   HF_TWO_67_33,
+   HF_THREE_33_34_33,
+   HF_THREE_25_25_50,
+   HF_THREE_25_50_25,
+   HF_THREE_50_25_25,
+   HF_THREE_40_20_40,
+   HF_FOUR_25_25_25_25,
+   LAST_HEADER_LAYOUT
+} HeaderLayout;
+
+typedef struct HeaderLayoutImpl_ {
+   unsigned int columns;
+   const unsigned int widths[4];
```

Comment:
Why not use `size_t` here?

Reply:
Using `uint8_t` now cause more than 255 header `columns` is unfeasible and `widths` is just a percentage number between 0 and 100.
The access wrapper now returns `size_t` so callers don't need to worry about the actual type.